tensorflow/lite/delegates/nnapi/nnapi_delegate.cc

```diff
@@ -655,7 +655,22 @@ NNMemory::NNMemory(const NnApi* nnapi, const char* name, size_t size) {
   if (name && size > 0) {
     nnapi_ = nnapi;
     byte_size_ = size;
+#ifdef __ANDROID__
     fd_ = nnapi_->ASharedMemory_create(name, size);
+#else
+    // For non-Android platforms ASharedMemory_create needs unique name to
+    // create a shared memory object (see nnapi_implementation.cc).
+    char shm_name_buffer[L_tmpnam];
+    if (tmpnam(shm_name_buffer) == nullptr) {
+      throw new std::runtime_error("NN Memory failed");
```

Comment:
The exception here is causing problems and blocking the merge.

Could you remove the exception?